spec/controllers/curator/shared/shared_formats_and_actions.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples 'shared_get', type: :controller do |include_ark_context: false, has_collection_methods: true, has_member_methods: true|
+  routes { Curator::Engine.routes }
+  specify { expect(format).to be_truthy.and be_a_kind_of(Symbol) }
+  specify { expect(params).to be_truthy.and be_a_kind_of(Hash) }
+  specify { expect(serialized_hash).to be_truthy.and be_a_kind_of(Hash) }
+
+  let(:expected_content_type) { Mime[format].to_str }
+
+  describe 'GET' do
+    describe "#index", if: has_collection_methods do
+      it "return a serialized array of json objects" do
+        get :index, params: params
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq(expected_content_type)
+        expect(json_response).to be_a_kind_of(Hash).and have_key(resource_key.pluralize)
+        expect(json_response).to include(resource_key.pluralize => include(a_hash_including(serialized_hash)))
+      end
+    end
+
+    describe "#show", if: has_member_methods do
```

Comment:
Do we want to add some examples for when an invalid/non-existent id is passed?

Reply:
Good point. I will add that case